src/main/java/ch/njol/skript/conditions/CondMatches.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * This file is part of Skript.
+ *
+ * Skript is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Skript is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+
+package ch.njol.skript.conditions;
+
+import java.util.Arrays;
+import java.util.regex.Pattern;
+
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+
+@Name("Matches")
+@Description("Test whether strings match defined regular expressions.")
```

Comment:
In my opinion, I feel this desperation is quite lacklustre. 
I know that you have regex matching available here, but there is absolutely no mention of it. I recommend updating the description to give more detail on what this condition actually does.

Reply:
It mentioned regular expressions though, do you mean explaining what regex is? If so the description would need to be way longer, and some people learn by reading the docs, others by watching YouTube tutorials, so doing that wouldn’t really be helpful, at most what could be done would be to link a few resources

Reply:
No, regex doesn't need to be explained, but it isn't mentioned at all. If someone where to open the docs and search for "regex", they're not going to find anything, since its not mentioned here.

Reply:
But regular expression and regex mean the same thing

Reply:
Not everyone knows that.... again, as I've said, if someone SEARCHES the word "regex" they aren't going to find any results.